File: Tyche.Android/session/src/main/java/com/felipearpa/tyche/session/authentication/domain/AuthenticationRepository.kt
package com.felipearpa.tyche.session.authentication.domain

import com.felipearpa.tyche.session.AccountBundle

interface AuthenticationRepository {
    suspend fun sendSignInLinkToEmail(email: String): Result<Unit>
    suspend fun signInWithEmailLink(email: String, emailLink: String): Result<ExternalAccountId>
    suspend fun logout(): Result<Unit>

    suspend fun linkAccount(accountLink: AccountLink): Result<AccountBundle>
}

File: Tyche.iOS/Session/Sources/Session/Authentication/Domain/AuthenticationRepository.swift
protocol AuthenticationRepository {
    func sendSignInLinkToEmail(email: String) async -> Result<Void, Error>
    func signInWithEmailLink(email: String, emailLink: String) async -> Result<ExternalAccountId, Error>
    func signInWithEmailAndPassword(email: String, password: String) async -> Result<ExternalAccountId, Error>
    func logout() async -> Result<Void, Error>
    func linkAccount(accountLink: AccountLink) async -> Result<AccountBundle, Error>
}
